### DISCUSSION TITLE
Is it possible to use boss with kitty +runpy

### DISCUSSION THREAD

**Discussion by jackielii** [kovidgoyal/kitty#6973] [Q&A] [answered]
So my question is: When I run:

```sh
kitty +runpy 'from kitty.fast_data_types import get_boss;print(get_boss())' 
```

I get `None`. So it it possible to get boss here or it's just not the way it works?

Longer version:

So I'm building a tmux "choose-tree" equivalent in Go. I use `kitty @ ls` to get most information out from kitty, but there are something that's not available and I want to run a one-off script to retrieve the information from `Boss` object.

My question is: is this possible? If not, is there another way?

**Comment by kovidgoyal** [comment, marked as answer]
No boss is only created in the actual kitty process. When you use runpy
you are running a second process. If you want access to the boss object
you do so by 

kitten @ kitten mykitten.py

This will run mykitten.py in the main kitty process context. It will
have access to boss in its handle_response() function.
